e2e test TestClusterGroup on kind is flaky

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
It failed twice for an irrelevant PR:
https://github.com/antrea-io/antrea/runs/5387722637?check_suite_focus=true
[e2e-kind-encap-no-proxy.tar.gz.zip](https://github.com/antrea-io/antrea/files/8174294/e2e-kind-encap-no-proxy.tar.gz.3.zip)
```
=== RUN   TestClusterGroup
    fixtures.go:194: Creating 'antrea-test' K8s Namespace
2022-03-02 08:01:24.431379 I | Applying Antrea YAML
2022-03-02 08:01:25.186036 I | Waiting for all Antrea DaemonSet Pods
2022-03-02 08:01:26.195865 I | Checking CoreDNS deployment
I0302 08:01:26.214493   19065 k8s_util.go:901] Creating/updating Pod 'x/a'
I0302 08:01:26.214532   19065 k8s_util.go:239] Creating/updating Deployment 'x/xa'
E0302 08:01:26.219956   19065 antreapolicy_test.go:101] deployments.apps "xa" not found
unable to create/update Deployment 'x/a'
I0302 08:01:26.228669   19065 k8s_util.go:943] Deleting test Namespace x
I0302 08:01:26.231914   19065 k8s_util.go:943] Deleting test Namespace y
E0302 08:01:26.233864   19065 k8s_util.go:945] Error when deleting Namespace 'y': namespaces "y" not found
I0302 08:01:26.233953   19065 k8s_util.go:943] Deleting test Namespace z
E0302 08:01:26.235670   19065 k8s_util.go:945] Error when deleting Namespace 'z': namespaces "z" not found
    antreapolicy_test.go:103: test failed: unable to create/update Deployment 'x/a': deployments.apps "xa" not found
    fixtures.go:269: Exporting test logs to '/home/runner/work/antrea/antrea/log/TestClusterGroup/beforeTeardown.Mar02-08-01-26'
    fixtures.go:396: Deleting 'antrea-test' K8s Namespace
--- FAIL: TestClusterGroup (10.77s)
```

https://github.com/antrea-io/antrea/runs/5387722979?check_suite_focus=true
[e2e-kind-noencap.tar.gz.zip](https://github.com/antrea-io/antrea/files/8174298/e2e-kind-noencap.tar.gz.1.zip)
```
=== RUN   TestClusterGroup
    fixtures.go:194: Creating 'antrea-test' K8s Namespace
2022-03-02 08:03:00.373544 I | Applying Antrea YAML
2022-03-02 08:03:01.078030 I | Waiting for all Antrea DaemonSet Pods
2022-03-02 08:03:02.085986 I | Checking CoreDNS deployment
I0302 08:03:02.107501   18997 k8s_util.go:901] Creating/updating Pod 'x/a'
I0302 08:03:02.107753   18997 k8s_util.go:239] Creating/updating Deployment 'x/xa'
E0302 08:03:02.111732   18997 antreapolicy_test.go:101] deployments.apps "xa" not found
unable to create/update Deployment 'x/a'
I0302 08:03:02.116822   18997 k8s_util.go:943] Deleting test Namespace x
I0302 08:03:02.118591   18997 k8s_util.go:943] Deleting test Namespace y
E0302 08:03:02.120059   18997 k8s_util.go:945] Error when deleting Namespace 'y': namespaces "y" not found
I0302 08:03:02.120084   18997 k8s_util.go:943] Deleting test Namespace z
E0302 08:03:02.122119   18997 k8s_util.go:945] Error when deleting Namespace 'z': namespaces "z" not found
    antreapolicy_test.go:103: test failed: unable to create/update Deployment 'x/a': deployments.apps "xa" not found
    fixtures.go:269: Exporting test logs to '/home/runner/work/antrea/antrea/log/TestClusterGroup/beforeTeardown.Mar02-08-03-02'
    fixtures.go:396: Deleting 'antrea-test' K8s Namespace
--- FAIL: TestClusterGroup (10.77s)
```

--- a/test/e2e/framework.go
+++ b/test/e2e/framework.go
@@ -30,6 +30,7 @@ import (
 	"time"
 
 	"github.com/containernetworking/plugins/pkg/ip"
+	log "github.com/sirupsen/logrus"
 	"golang.org/x/mod/semver"
 	"gopkg.in/yaml.v2"
 	appsv1 "k8s.io/api/apps/v1"
@@ -622,6 +623,13 @@ func (data *TestData) deleteNamespace(namespace string, timeout time.Duration) e
 		GracePeriodSeconds: &gracePeriodSeconds,
 		PropagationPolicy:  &propagationPolicy,
 	}
+
+	// To log time statistics
+	startTime := time.Now()
+	defer func() {
+		log.Infof("Deleting Namespace %s took %v", namespace, time.Since(startTime))
+	}()
+
 	if err := data.clientset.CoreV1().Namespaces().Delete(context.TODO(), namespace, deleteOptions); err != nil {
 		if errors.IsNotFound(err) {
 			// namespace does not exist, we return right away

--- a/test/e2e/k8s_util.go
+++ b/test/e2e/k8s_util.go
@@ -941,7 +941,7 @@ func (k *KubernetesUtils) Cleanup(namespaces []string) {
 
 	for _, ns := range namespaces {
 		log.Infof("Deleting test Namespace %s", ns)
-		if err := k.clientset.CoreV1().Namespaces().Delete(context.TODO(), ns, metav1.DeleteOptions{}); err != nil {
+		if err := k.deleteNamespace(ns, defaultTimeout); err != nil {
 			log.Errorf("Error when deleting Namespace '%s': %v", ns, err)
 		}
 	}

